Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -384,4 +356,269 @@ static void _GetGlyphRunDetails(_CTTypesetter* ts, CFRange range, CGRect frameSi
+    THROW_IF_FAILED(dwriteFactory->GetSystemFontCollection(&fontCollection));
+
+    // Get the number of font families in the collection.
+    uint32_t count = 0;
+    count = fontCollection->GetFontFamilyCount();
+
+    for (uint32_t i = 0; i < count; ++i) {
+        // Get the font family.
+        ComPtr<IDWriteFontFamily> fontFamily;
+        THROW_IF_FAILED(fontCollection->GetFontFamily(i, &fontFamily));
+
+        // Get a list of localized strings for the family name.
+        ComPtr<IDWriteLocalizedStrings> familyNames;
+        THROW_IF_FAILED(fontFamily->GetFamilyNames(&familyNames));
+
+        [fontFamilyNames addObject:_ConvertLocalizedStringToNSString(familyNames.Get())];
+    }
+
+    return fontFamilyNames;
+}
+
+NSArray<NSString*>* _DWriteGetFontNamesForFamilyName(NSString* familyName) {
+    NSMutableArray<NSString*>* fontNames = [NSMutableArray<NSString*> array];
+
+    ComPtr<IDWriteFactory> dwriteFactory = __GetDWriteFactoryInstance();
+
+    // Get the system font collection.
+    ComPtr<IDWriteFontCollection> fontCollection;
+    THROW_IF_FAILED(dwriteFactory->GetSystemFontCollection(&fontCollection));
+
+    // Get the font family.
+    UINT32 index = 0;
+    BOOL exists = false;
+
+    NSUInteger familyNameLength = familyName.length;
+    std::vector<UniChar> unicharFamilyName(familyNameLength + 1);
+    [familyName getCharacters:unicharFamilyName.data() range:NSMakeRange(0, familyNameLength)];
+
+    THROW_IF_FAILED(fontCollection->FindFamilyName(reinterpret_cast<wchar_t*>(unicharFamilyName.data()), &index, &exists));
+    if (!exists) {
+        TraceError(TAG, L"Failed to find the font family name.");
+        return fontNames;
+    }
+
+    ComPtr<IDWriteFontFamily> fontFamily;
+    THROW_IF_FAILED(fontCollection->GetFontFamily(index, &fontFamily));
+
+    ComPtr<IDWriteFontList> fontList;
+    THROW_IF_FAILED(
+        fontFamily->GetMatchingFonts(DWRITE_FONT_WEIGHT_THIN, DWRITE_FONT_STRETCH_UNDEFINED, DWRITE_FONT_STYLE_NORMAL, &fontList));
+
+    UINT32 count = 0;
+    count = fontList->GetFontCount();
+
+    for (UINT32 i = 0; i < count; i++) {
+        ComPtr<IDWriteFont> font;
+        THROW_IF_FAILED(fontList->GetFont(i, &font));
+
+        ComPtr<IDWriteLocalizedStrings> fullName;
+        BOOL exist = FALSE;
+        THROW_IF_FAILED(font->GetInformationalStrings(DWRITE_INFORMATIONAL_STRING_FULL_NAME, &fullName, &exist));
+
+        if (exist) {
+            NSString* name = _ConvertLocalizedStringToNSString(fullName.Get());
+            if (name.length == 0) {
+                TraceError(TAG, L"Failed to convert the localized string to wide string.");
+                return fontNames;
+            }
+
+            [fontNames addObject:name];
+        }
+    }
+
+    return fontNames;
+}
+
+NSString* _DWriteGetFamilyNameForFontName(CFStringRef fontName) {
+    static NSDictionary<NSString*, NSString*>* fontToFamilyMap = nil;
+
+    static dispatch_once_t initOnce = 0;
+    dispatch_once(&initOnce,
```

Comment:
I did not find one while looking for it. Put a note here and I don't think this is a mad assumption to make

---
In reply to: [78232112](https://github.com/Microsoft/WinObjC/pull/987#discussion_r78232112) [](ancestors = 78232112,78218544)